### PR TITLE
Add sample JWT generator project

### PR DESCRIPTION
**Pull request by HeshanSudarshana**
### Purpose
Adding a sample JWT generator project to the samples folder. This sample project is to be used to refer in the documentation of backend JWT generation.

### Issues
Task

### Automation tests
- Unit tests added: No
- Integration tests added: No

### Tested environments
Ubuntu 20.04

---

#### Maintainers: Check before merge

- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: samples/sample-jwt-generator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <version>1.0-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.am.microgw</groupId>
+            <artifactId>mgw-jwt-generator</artifactId>
+            <version>3.2.0</version>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (praminda) [inline samples/sample-jwt-generator/pom.xml:18]:
```suggestion
</project>

```

File: samples/sample-jwt-generator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Meruja) [inline samples/sample-jwt-generator/pom.xml:3]: Shall we add the year here?

File: samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            claims.put("scopes", (customClaims.get("scopes")));
+        }
+        return claims;
+    }
+
+    @Override
+    public Map<String, Object> populateCustomClaims(Map<String, Object> jwtInfo, ArrayList<String> restrictedClaims) {
+        Map<String, Object> claims = new HashMap();
+        for (String key: jwtInfo.keySet()) {
+            if (key.equals("customClaims")) {
```
Comment (tgtshanika) [inline samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java:84]: Please format the code

File: samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                Map<String, Object> customClaims = (Map<String, Object>) jwtInfo.get(key);
+                for (String subKey: customClaims.keySet()) {
+                    if (!restrictedClaims.contains(subKey)) {
```
Comment (tgtshanika) [inline samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java:87]: formatting issue

File: samples/sample-jwt-generator/pom.xml (hunk 2 above)
Comment (HeshanSudarshana) [reply samples/sample-jwt-generator/pom.xml:3]: Added with https://github.com/wso2/product-microgateway/pull/1613/commits/45bcd3462390fb8a500155b1accc2e40b6f75444

File: samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java (hunk 3 above)
Comment (HeshanSudarshana) [reply samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java:84]: Fixed with https://github.com/wso2/product-microgateway/pull/1613/commits/45bcd3462390fb8a500155b1accc2e40b6f75444

File: samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java (hunk 4 above)
Comment (HeshanSudarshana) [reply samples/sample-jwt-generator/src/main/java/sample/jwt/generator/SampleJWTGenerator.java:87]: Fixed with https://github.com/wso2/product-microgateway/pull/1613/commits/45bcd3462390fb8a500155b1accc2e40b6f75444

**Pull request merged**